### PR TITLE
Update compiler version for CI to 0.6.18

### PR DESCRIPTION
**Pull request by kritzcreek**

### CODE REVIEW

**Comment by jzxchiang1** [comment]
LGTM, it passes for me locally as well.

**Review by crusso — APPROVED** [review]

**Review by crusso — APPROVED** [review]
Rubber stamping

**Comment by kritzcreek** [comment]
> Rubber stamping

Thanks, but I'd actually need you to use your merge privilege as well ;)

**Comment by timohanke** [comment]
Why not 0.6.18?

File: Makefile [hunk 1]
```diff
@@ -9,4 +9,4 @@ check-format:
 	dhall type --file package-set.dhall;
 	@echo checked dhall files are well-typed.
 ci: check-format
-	vessel verify --version 0.6.7
+	vessel verify --version 0.6.14
```
Comment (kritzcreek) [inline Makefile:12]:
```suggestion
	vessel verify --version 0.6.18
```

**Comment by kritzcreek** [comment]
> Why not 0.6.18?

Because this was prompted by https://forum.dfinity.org/t/do-people-actually-use-vessel/9849, which mentioned 0.6.14 was the "latest" at the time.

But we should totally be able to update to 0.6.18 here. This just changes the version used to verify the package set on CI it doesn't have any impact for downstream users.

Updated the PR

**Comment by kritzcreek** [comment]
I've updated `base` which reveals [deprecation warnings](https://github.com/dfinity/vessel-package-set/runs/4689398598?check_suite_focus=true#step:6:1) around `Option.unwrap` and `Array.append` in these libraries:
- `matchers`
- `scc`
- `base32`

Not a blocker, but it would be good to PR updated to these libraries so they compile without warnings again. I can take care of `matchers` later this week.

**Comment by timohanke** [comment]
Created issues in the repos of scc and base32.

**Comment by kritzcreek** [comment]
Since they're just warnings we can easily release a new version of the package-set like this and just make new releases once those libraries are updated.

**Comment by timohanke** [comment]
Ok, let's do that. scc is solved by now. base32 still pending but it turns out that base32 itself uses the deprecated functions only inside its own tests.

**Review by crusso — APPROVED** [review]
LGTM

**Review by crusso — APPROVED** [review]
LTMR, really.

**Review by crusso — APPROVED** [review]
Really, really approve

**Review by timohanke — APPROVED** [review]

**Pull request merged**